app/src/main/java/com/github/se/cyrcle/ui/list/ListScreen.kt

```diff
@@ -269,26 +295,48 @@ fun FilterHeader(parkingViewModel: ParkingViewModel) {
 fun FilterSection(
     title: String,
     expandedState: MutableState<Boolean>,
+    onReset: () -> Unit,
+    onApply: () -> Unit,
     content: @Composable () -> Unit
 ) {
   Column(
       modifier =
-          Modifier.padding(8.dp)
+          Modifier.padding(vertical = 8.dp)
               .border(1.dp, Color.Gray, shape = MaterialTheme.shapes.medium)
               .background(
                   MaterialTheme.colorScheme.surfaceBright, shape = MaterialTheme.shapes.medium)
               .padding(8.dp)) {
-        Text(
-            text = title,
-            style = MaterialTheme.typography.titleMedium,
-            modifier =
-                Modifier.clickable(onClick = { expandedState.value = !expandedState.value })
-                    .padding(8.dp)
-                    .fillMaxWidth(),
-            testTag = title)
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically) {
+              ClickableText(
+                  text = AnnotatedString(stringResource(R.string.list_screen_clear)),
+                  onClick = { onReset() },
+                  style =
+                      MaterialTheme.typography.labelSmall.copy(
+                          color = MaterialTheme.colorScheme.primary),
+                  modifier =
+                      Modifier.padding(horizontal = 4.dp).testTag("Clear" + title + "Button"))
+              Text(
+                  text = title,
+                  style = MaterialTheme.typography.titleMedium,
+                  modifier =
+                      Modifier.clickable(onClick = { expandedState.value = !expandedState.value })
+                          .padding(6.dp),
+                  testTag = title)
+              ClickableText(
+                  text = AnnotatedString(stringResource(R.string.list_screen_apply)),
+                  onClick = { onApply() },
+                  style =
+                      MaterialTheme.typography.labelSmall.copy(
+                          color = MaterialTheme.colorScheme.primary),
+                  modifier =
+                      Modifier.padding(horizontal = 4.dp).testTag("Apply" + title + "Button"))
```

Comment:
Here you could have used the string format `"Apply$titleButton"` instead of concatenation

Reply:
You're right, I didn't overthink it as it was "just" a testTag. Turns out I underthinked it